src/common/transformations/src/ov_ops/vl_sdpa.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ov_ops/vl_sdpa.hpp"
+
+#include "augru_sequence_shape_inference.hpp"
+#include "itt.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "ov_ops/augru_sequence.hpp"
+#include "scaled_dot_product_attention_shape_inference.hpp"
+namespace ov {
+namespace op {
+namespace internal {
+
+VLSDPA::VLSDPA(const OutputVector& inputs,
+               const std::vector<int64_t>& order_q,
+               const std::vector<int64_t>& order_k,
+               const std::vector<int64_t>& order_v,
+               const std::vector<int64_t>& order_out)
+    : Op(inputs),
+      m_order_q(order_q),
+      m_order_k(order_k),
+      m_order_v(order_v),
+      m_order_out(order_out) {
+    constructor_validate_and_infer_types();
```

Comment:
Should we validate transpose orders with respect to the input ranks?

Reply:
How about - "void VLSDPA::validate_and_infer_types()" ?